python/cudf/cudf/tests/test_df_protocol.py

```diff
@@ -229,3 +268,12 @@ def test_NA_mixed_dtype():
         string=[None, None, None, "df protocol", None],
     )
     assert_df_unique_dtype_cols(data_mixed)
+
+
+@pytest.mark.skipif(
+    not PANDAS_GE_150,
+    reason="Pandas versions < 1.5.0 do not support interchange protocol",
+)
+def test_from_cpu_df(pandas_df):
+    df = pd.DataFrame({"a": [1, 2, 3], "b": ["x", "y", "z"]})
+    cudf.from_dataframe(df, allow_copy=True)
```

Comment:
Thanks! I added some null tests.

> Also can we check the round-trip in the opposite direction? (Assuming that is supported too).

Today, that segfaults because Pandas doesn't check for the device type of the incoming buffers and tries to interpret the buffer pointers as host pointers :(